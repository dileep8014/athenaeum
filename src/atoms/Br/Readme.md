## What is the `<Br />` tag?
`<Br />` is a `<span />` that acts like a line-break to allow line-breaking based on media query sizes.

Example:
`<Br small large ... />`

### Br Example:
```jsx
  <div>
    <h2>I break on ↵<Br large />large.</h2>
    <h2>I break on ↵<Br small medium large />medium large.</h2>
    <h3>I break only on ↵<Br medium />medium.</h3>
    <h3>I break on ↵<Br medium small />medium & small.</h3>

    <h5>I break only on ↵<Br small />small</h5>

  </div>
```
