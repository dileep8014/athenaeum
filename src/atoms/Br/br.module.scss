$breakpoints: (
  $small-only: 'sm',
  $medium-only: 'med',
  $large-only: 'lg',
  $xlarge-only: 'x-lg',
  $xxlarge-only: 'xx-lg'
);

.br {
  display: run-in;
}

@each $breakpoint, $class in $breakpoints {
  @media #{$breakpoint} {
    .#{$class} {
      display: block;
    }
  }
}
