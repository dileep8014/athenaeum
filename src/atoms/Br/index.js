import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './br.module.scss';

function Br(props) {
  const {
    small,
    medium,
    large,
    xLarge,
    xxLarge,
    className,
  } = props;

  const classes = [
    styles['br'],
    small && styles['sm'],
    medium && styles['med'],
    large && styles['lg'],
    xLarge && styles['x-lg'],
    xxLarge && styles['xx-lg'],
    className,
  ];

  return <span className={classnames(...classes)} />;
}


Br.propTypes = {
  /**
   * small breakpoint
   */
  small: PropTypes.bool,

  /**
   * medium breakpoint
   */
  medium: PropTypes.bool,

  /**
   * large breakpoint
   */
  large: PropTypes.bool,

  /**
   * X-large breakpoint
   */
  xLarge: PropTypes.bool,

  /**
   * XX-Large breakpoint
   */
  xxLarge: PropTypes.bool,

  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
};

Br.defaultProps = {
  // Place any default props here.
};

export default Br;
