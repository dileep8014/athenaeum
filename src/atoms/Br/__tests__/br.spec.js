import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Br from 'atoms/Br';

describe('<Br />', () => {

  it('renders correctly', () => {
    const wrapper = mount(<Br />);

    expect(wrapper.type()).toEqual(Br);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<Br />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
