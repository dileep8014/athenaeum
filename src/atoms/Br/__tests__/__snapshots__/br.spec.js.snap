// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Br /> renders correctly 1`] = `
<span
  className="br"
/>
`;
